### PR TITLE
feat(Designer): Added node selection functionality options

### PR DESCRIPTION
**Pull request by rllyy97**
## Main Changes
- Added option to suppress default functionality for selecting nodes in designer
  - In EiPaaS we are selecting nodes but opening a completely different panel, and not show the parameters panel like normal
- Added option to give an extra callback to run on selecting nodes in designer
  - This might be helpful for EiPaas work to open their new panel

### CODE REVIEW

File: libs/designer/src/lib/ui/CustomNodes/OperationCardNode.tsx [hunk 1]
```diff
@@ -74,6 +79,9 @@ const DefaultNode = ({ targetPosition = Position.Top, sourcePosition = Position.
   const isSecureInputsOutputs = useSecureInputsOutputs(id);
   const { status: statusRun, error: errorRun, code: codeRun, repetitionCount } = runData ?? {};
 
+  const suppressDefaultNodeSelect = useSuppressDefaultNodeSelectFunctionality();
+  const nodeSelectCallbackOverride = useAddedNodeSelectCallback();
```
Comment (ccastrotrejo) [inline libs/designer/src/lib/ui/CustomNodes/OperationCardNode.tsx:83]: NIT: I don't know if `useAddedNodeSelectCallback` is the right name. I am just picky about the `Added` part

File: libs/designer/src/lib/core/DesignerProvider.tsx [hunk 2]
```diff
@@ -22,15 +22,7 @@ const OptionsStateSet = ({ options, children }: any) => {
...
-    dispatch(
-      initDesignerOptions({
-        readOnly: options.readOnly,
-        isMonitoringView: options.isMonitoringView,
-        isDarkMode: options.isDarkMode,
-        useLegacyWorkflowParameters: options.useLegacyWorkflowParameters,
-        isXrmConnectionReferenceMode: options.isXrmConnectionReferenceMode,
-      })
-    );
+    dispatch(initDesignerOptions(options));
```
Comment (ccastrotrejo) [inline libs/designer/src/lib/core/DesignerProvider.tsx:25]: Do we need to update the designerOptionsSlice for the suppressDefaultNodeSelectFunctionality and addedNodeSelectCallback?

**Review by ccastrotrejo — CHANGES_REQUESTED** [review]

File: libs/designer/src/lib/core/DesignerProvider.tsx (hunk 2 above)
Comment (rllyy97) [reply libs/designer/src/lib/core/DesignerProvider.tsx:25]: Switching it over to this implementation makes it so we don't have to specify each option anymore

File: libs/designer/src/lib/ui/CustomNodes/OperationCardNode.tsx (hunk 1 above)
Comment (rllyy97) [reply libs/designer/src/lib/ui/CustomNodes/OperationCardNode.tsx:83]: Yeah I wasn't really sure how to phrase both of these, do you have any suggestions?

File: libs/designer/src/lib/core/DesignerProvider.tsx (hunk 2 above)
Comment (ccastrotrejo) [reply libs/designer/src/lib/core/DesignerProvider.tsx:25]: Yep, thats great, but I wonder if on the execution of `initDesignerOptions` we need to specify this new properties.
image

File: libs/designer/src/lib/ui/CustomNodes/OperationCardNode.tsx (hunk 1 above)
Comment (rllyy97) [reply libs/designer/src/lib/ui/CustomNodes/OperationCardNode.tsx:83]: Discussed offline, changing to `nodeSelectAdditionalCallback`

**Review by ccastrotrejo — APPROVED** [review]

**Pull request merged**